src/lib/onboarding/service.ts

```diff
@@ -32,6 +38,16 @@ export const validateAccount = (payload: any): payload is AccountToCreate => {
   );
 };
 
+export const validateAccountWithResponse = (
+  payload: any,
+): payload is AccountToCreateWithResponse => {
+  if (!validateAccount(payload)) return false;
+
+  const _payload = payload as any;
```

Comment:
This isn't much better, but you won't need to cast.
```suggestion
  const _payload: Record<string, any> = payload;
```